### ISSUE TITLE
Update dialog component to Material 3

### ISSUE THREAD

**Issue by matthiasdebaat** [home-assistant/frontend#13084]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Material Design has introduced version 3 of its [dialogs](https://m3.material.io/components/dialogs/overview).

### Describe the behavior you expected

There are two types of dialogs:

**Basic dialog**
Basic dialogs interrupt users with urgent information, details, or actions. Common use cases for basic dialogs include alerts, quick selection, and confirmation. 

**Full-screen dialog**
Full-screen dialogs fill the entire screen, containing actions that require a series of tasks to complete. One example is creating a calendar entry with the event title, date, location, and time.

Some highlights of its guidelines:
- A dialog’s purpose should be communicated by its headline and buttons.
- Dialogs should contain a maximum of two actions, for example `Save` and `Cancel`.
- Confirmation dialogs doesn't contain a close icon in the header.
- (Mobile) full screen dialogs have the close icon on the left, on large screens it's on the right.
- Dialogs can reach a maximum height and width of 560px on large screens.
- When a dialog reaches the maximum height, the content can be vertical scrollable. The dialog title and buttons remain fixed.

![dialogs](https://user-images.githubusercontent.com/20203284/177333155-9d2cacce-03ac-4efa-8c25-3d7e1d9949d6.png)

### Steps to reproduce the issue

1.

### What version of Home Assistant Core has the issue?

core-2022.6.7

**Comment by steverep** [comment]
Material three has only been released for android so far. By the looks of the web components repository, it's in work but might be a while.

**Comment by matthiasdebaat** [comment]
> Material three has only been released for android so far. By the looks of the web components repository, it's in work but might be a while.

@bramkragten did we create our own dialog or did we use Materials component? If we created our own, we can restyle it to the latest design.

**Comment by steverep** [comment]
I'm not sure if it's worth a separate issue, but the inconsistent use of close and cancel buttons also needs to be fixed for both accessibility and general usability.  Apart from small confirmation and acknowledgement dialogs, some larger dialogs have both close and cancel buttons, but some have only one or the other.

#14151 also deals with inconsistency, but specifically for escape key and scrim clicks.
